util/log/log.go

```diff
@@ -34,74 +43,169 @@ func FatalOnPanic() {
 	}
 }
 
+func logKV(buf *bytes.Buffer, kvs ...[]interface{}) {
+	var i int
+	var s string
+	var kv []interface{}
+	for kvi := range kvs {
+		kv = kvs[kvi]
+		l := len(kv)
+		for i = 0; i < l; i++ {
+			if i%2 == 0 {
+				if kvi != 0 || i != 0 {
+					buf.WriteString(" ")
+				}
+			} else {
+				buf.WriteString("=")
+			}
+			switch kt := kv[i].(type) {
+			case string:
+				s = kt
+			case fmt.Stringer:
+				s = kt.String()
+			default:
+				s = fmt.Sprintf("%v", kt)
+			}
+			if i%2 == 0 {
+				buf.WriteString(s)
+			} else {
+				buf.WriteString(strconv.Quote(s))
+			}
+		}
+		if i%2 != 0 {
```

Comment:
I'm also not convinced this should be a panic - it could just add an "extra argument" warning to the message instead, similar to how functions like Printf handle missing arguments.